src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3698,15 +3699,18 @@ void func_80836BEC(Player* this, PlayState* play) {
             }
 
             if (this->focusActor != NULL) {
-                if ((this->actor.category == ACTORCAT_PLAYER) && (this->focusActor != this->unk_684) &&
+                if ((this->actor.category == ACTORCAT_PLAYER) && (this->focusActor != this->autoLockOnActor) &&
                     Attention_ShouldReleaseLockOn(this->focusActor, this, ignoreLeash)) {
                     func_8008EDF0(this);
                     this->stateFlags1 |= PLAYER_STATE1_LOCK_ON_FORCED_TO_RELEASE;
                 } else if (this->focusActor != NULL) {
                     this->focusActor->attentionPriority = 40;
                 }
-            } else if (this->unk_684 != NULL) {
-                this->focusActor = this->unk_684;
+            } else if (this->autoLockOnActor != NULL) {
+                // `autoLockOnActor` does not take precedence over `focusActor` if it already exists.
+                // However, `autoLockOnActor` is expected to be set with `Player_SetAutoLockOnActor`
+                //  which will release any existing lock-on. So it effectively does take precedence.
```

Comment:
I'm not sure how to read this, could you reword? What is `it` in the first sentence? And since this is overriding focusActor with autoLockOnActor if autoLockOnActor exists, doesn't that mean that autoLockOnActor takes precedence here too?

Reply:
"it" is `focusActor` 
if you look at the surrounding if/if-else youll see that player first prioritizes `focusActor` if it already exists. Only if it doesnt will `focusActor` get set to `autoLockOnActor`.

In other words, `autoLockOnActor` itself does not force itself to be the main focus if it is set. You need to use the function to release a lock on first in order to "force" this one to be used

Reply:
> if you look at the surrounding if/elseif youll see that player first prioritizes focusActor if it already exists
I see thanks, maybe move this comment closer to here, or mention the previous if in the comment?